python/strings_udf/strings_udf/__init__.py

```diff
@@ -41,59 +25,70 @@ def _get_appropriate_file(sms, cc):
         return None
 
 
-# adapted from PTXCompiler
-cp = subprocess.run([sys.executable, "-c", CMD], capture_output=True)
-if cp.returncode == 0:
-    # must have a driver to proceed
-    versions = [int(s) for s in cp.stdout.strip().split()]
-    driver_version = tuple(versions[:2])
-    runtime_version = tuple(versions[2:])
+def maybe_patch_numba_linker(driver_version):
+    # Numba thinks cubinlinker is only needed if the driver is older than the ctk
+    # but when strings_udf is present, it might also need to patch because the PTX
+    # file strings_udf relies on may be newer than the driver as well
+    if driver_version < STRINGS_UDF_PTX_VERSION:
+        logger.debug(
+            "Driver version %s.%s needs patching due to strings_udf"
+            % driver_version
+        )
+        if _numba_version_ok:
+            logger.debug("Patching Numba Linker")
+            Linker.new = new_patched_linker
+        else:
+            logger.debug("Cannot patch Numba Linker - unsupported version")
+
+
+def _get_ptx_file():
+    if "RAPIDS_NO_INITIALIZE" in os.environ:
+        cc = int(os.environ.get("STRINGS_UDF_CC", "52"))
```

Comment:
The idea here is that downstream consumers (like `dask-cuda`) that set `RAPIDS_NO_INITIALIZE` have a way to advertise the compute capability of the device they are running on (without creating a context). Do we care about bikeshedding this name?